cost/cost_test.go

```diff
@@ -0,0 +1,45 @@
+package cost_test
+
+import (
+	"testing"
+
+	"github.com/cycloidio/terracost/cost"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewHourly(t *testing.T) {
+	c := cost.NewHourly(decimal.NewFromFloat(1.23))
+	assertDecimalEqual(t, decimal.NewFromFloat(897.9), c.Decimal) // 897.9 = 1.23 * 730
+}
+
+func TestNewMonthly(t *testing.T) {
+	c := cost.NewMonthly(decimal.NewFromFloat(1.23))
+	assertDecimalEqual(t, decimal.NewFromFloat(1.23), c.Decimal)
+}
+
+func TestCost_Hourly(t *testing.T) {
+	c := cost.NewMonthly(decimal.NewFromFloat(897.9))
+	assertDecimalEqual(t, decimal.NewFromFloat(1.23), c.Hourly())
+}
+
+func TestCost_Monthly(t *testing.T) {
+	c := cost.NewHourly(decimal.NewFromFloat(1.23))
+	assertDecimalEqual(t, decimal.NewFromFloat(897.9), c.Monthly()) // 897.9 = 1.23 * 730
+}
+
+func TestCost_Add(t *testing.T) {
+	c1 := cost.NewMonthly(decimal.NewFromFloat(1.23))
+	c2 := cost.NewMonthly(decimal.NewFromFloat(3.21))
+	assertDecimalEqual(t, decimal.NewFromFloat(4.44), c1.Add(c2).Decimal)
+}
+
+func TestCost_MulDecimal(t *testing.T) {
+	c := cost.NewMonthly(decimal.NewFromFloat(1.23))
+	d := decimal.NewFromInt(3)
+	assertDecimalEqual(t, decimal.NewFromFloat(3.69), c.MulDecimal(d).Decimal)
+}
+
+func assertDecimalEqual(t *testing.T, expected, actual decimal.Decimal) {
+	assert.Truef(t, expected.Equal(actual), "Not equal:\nexpected: %s\nactual  : %s", expected, actual)
+}
```

Comment:
I assume `decimal.Decimal` cannot be compared with `assert.Equal`?

Reply:
Yes, the internal representation is not suitable for comparison, the only way to accurately compare them is to use `Decimal.Equal`